flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamTest.java

```diff
@@ -906,6 +919,256 @@ public void testChannelSelectors() {
 	}
 
 	/////////////////////////////////////////////////////////////
+	// KeyBy testing
+	/////////////////////////////////////////////////////////////
+
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
```

Comment:
The thing is that if I remember correctly we discussed about how to check for exceptions earlier this year. And if I remember correctly the outcome was sth along the lines of using JUnit facilities. Feel free to correct me on this one.